src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -190,6 +190,29 @@ final case class AgreementApiServiceImpl(
     )
   }
 
+  override def rejectAgreement(agreementId: String, payload: AgreementRejectionPayload)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerAgreement: ToEntityMarshaller[Agreement]
+  ): Route = authorize(ADMIN_ROLE) {
```

Comment:
Just an admin can refuse an agreement? Is has to be designed like that?

Reply:
Right now only admins can change the state of an agreement (activate, suspend, submit, upgrade), so I would say `yes`. But not sure if you had any different use case in mind

Reply:
No, no, I was just asking